algorithms/linfa-linear/src/ols.rs

```diff
@@ -9,12 +9,11 @@ use linfa_linalg::qr::LeastSquaresQrInto;
 use ndarray::{concatenate, s, Array, Array1, Array2, ArrayBase, Axis, Data, Ix1, Ix2};
 #[cfg(feature = "blas")]
 use ndarray_linalg::LeastSquaresSvdInto;
-use serde::{Deserialize, Serialize};
 
 use linfa::dataset::{AsSingleTargets, DatasetBase};
 use linfa::traits::{Fit, PredictInplace};
 
-#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
```

Comment:
Removing the mandatory `serde` dependency is great, but I'd also like to have `serde` support as an optional dependency. Can you introduce an optional `serde` feature into the affected crates like it's done in `linfa-clustering`? Also, instead of removing the `Serialize, Deserialize` annotations like it's done here, feature gate them behind the `serde` feature. You should also add the `linfa-linear/serde` and `linfa-logistic/serde` features to the Serde checking step in `.github/workflows/checking.yml`.